plugins/module_utils/aci.py

```diff
@@ -1286,6 +1286,8 @@ def get_diff(self, aci_class):
             # add name back to config only if the configs do not match
             if config:
                 # TODO: If URLs are built with the object's name, then we should be able to leave off adding the name back
+                if required_properties and isinstance(required_properties, dict):
```

Comment:
Why is the name of the key_ring (from the module) being explicitly updated here? Shouldn't the comparison above take care of it?

Reply:
Correct the diff takes care of this and removes the attribute "name" of the key ring correctly because it is already existing. This attribute is a mandatory attribute and the request will fail when not provided. If there is a config change to a keyring, for instance "description" attribute change. Then the "name" attribute remains the same, gets deleted from the config, and the request fails.

I have considered doing a name specific solution but do not know if there will be other variations of this behaviour. Chose for a dict of key/value pairs, because we have other identifier attributes in ACI, like "ip" for instance. Also it could be possible that there are cases where the amount of identifier mandatory attributes is higher than 1.